src/Orc.SystemInfo/Wmi/WindowsManagementObject.cs

```diff
@@ -64,24 +62,24 @@ internal WindowsManagementObject(IWbemClassObject wmiObject)
             }
         }
 
-        public void Dispose()
+        protected override void DisposeUnmanaged()
         {
-            if (!_disposed)
+            base.DisposeUnmanaged();
+            if(_wbemClassObject is not null)
```

Comment:
should we set _wbemClassObject to null? after releasing the COM object?

Reply:
Catel already throws an exception if things are already disposed. It's hidden in the implementation, so it's indeed not clear here. I think we could set it to null .However, this could potentially lead to NullReferenceExceptions (which could cause weird situations). I'd rather have all calls access this field checking for disposed.

Reply:
In general it's good idea if you want to force GC free as soon as possible for big objects, but not necessary (according to msdn)